src/cryptsetup/cryptsetup-generator.c

```diff
@@ -152,7 +152,9 @@ static int create_disk(
                               f);
         } else
                 fprintf(f,
-                        "RequiresMountsFor=%s\n",
+                        "RequiresMountsFor=%s\n"
+                        "Requires=systemd-tmpfiles-setup-dev.service\n"
+                        "After=systemd-tmpfiles-setup-dev.service\n",
                         u_escaped);
```

Comment:
hmm, so I think we should merge it like this, but could you add a brief comment before this fprintf() that explains why we need systemd-tmpfiles-setup-dev? This is not obvious after all, and the next one reading this really should get an explanation

Reply:
done